specs/AppLifecycle/Instancing/AppLifecycle SingleMulti-Instancing.md

```diff
@@ -0,0 +1,422 @@
+    redirection. An app that has unregistered itself is not included in the list of instance that is
+    made available to the app.
+-   **Reunion**: Unregistering a key does not prevent an instance from being a target for
+    redirection. Nor does it exclude the instance from the list.
+
+### Activation Events
+
+In order to handle reactivation, the app can register for an Activated event.
+
+-   **Platform**: The event passes a `Windows.ApplicationModel.Activation.IActivatedEventArgs` to
+    the app.
+-   **Reunion**: The event passes a `Microsoft.Windows.AppLifecycle.ActivationArguments` instance to
+    the app, which contains one of the `-ActivatedEventArgs` instances.
+
+## Examples
+
+### Redirection logic based on activation kind
+
+In this example, the app calls GetActivatedEventArgs, and if it has been activated as a result of a
+file-type association, it checks first to see if there is any other instance already running that
+has this file open. The app can tell whether the instance has the file open because it includes the
+filename as part of the key that it registers with the system.
+
+If it finds another instance with the file open, the app redirects this activation to that instance,
+and then does nothing. After redirecting the activation to the other instance, the system will then
+terminate this instance.
+
+If it doesn't find another instance with the file open, it registers itself as the instance that has
+this file open, and continues with normal app initialization (creating its main window, starting its
+message pump, and so on).
+
+```c++
+int APIENTRY wWinMain(
+    _In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance,
+    _In_ LPWSTR lpCmdLine, _In_ int nCmdShow)
+{
+    // First, we'll get our rich activation event args.
+    ActivationArguments activationArgs =
+        AppInstance::GetCurrent().GetActivatedEventArgs();
+
+    // An app might want to set itself up for possible redirection in
+    // the case where it opens files - for example, to prevent multiple
+    // instances from working on the same file.
+    ExtendedActivationKind kind = activationArgs.Kind;
+    if (kind == ExtendedActivationKind::File)
+    {
+        auto fileArgs = activationArgs.Data.as<FileActivatedEventArgs>();
+        IStorageItem file = fileArgs.Files().GetAt(0);
+
+        // Let's try to register this instance for this file.
+        AppInstance instance =
+            AppInstance::FindOrRegisterForKey(file.Name());
+        if (instance.IsCurrent)
+        {
+            // If we successfully registered this instance, we can now just
+            // go ahead and do normal initialization.
+            RegisterClassAndStartMessagePump(hInstance, nCmdShow);
+        }
+        else
+        {
+            // Some other instance has already registered for this file,
+            // so we'll redirect this activation to that instance instead.
+            instance.RedirectTo(activationArgs);
```

Comment:
How is the terminal-ness of RedirectTo manifested? Does the function never return? Would this bypass destructors?

Reply:
Added further description: in Reunion, redirection is not terminal.